bash_conditionals_loops/bash_conditionals_loops.md

```diff
@@ -0,0 +1,620 @@
+
+|Code| Meaning|
+|:-:| :-|
+| `[ -e FILE ]` | FILE exists |
+| `[ -s FILE ]` | FILE exists and is non-empty|
+|`[ STRING1 = STRING2 ]` | STRING1 is equal (identical) to STRING2 |
+|`[ INTEGER1 -eq INTEGER2 ]` | INTEGER1 is equal (numerically) to INTEGER2|
+|`[ INTEGER1 -gt INTEGER2 ]` | INTEGER1 is greater than  INTEGER2|
+|`[ INTEGER1 -lt INTEGER2 ]` | INTEGER1 is less than INTEGER2|
+
+<div class = "learnmore">
+
+As you use them more, you will learn more and more ways to combine and mix mathematical statements to make new ones. An exclamation point `!` lets you negate a statement, and the double brackets we saw in the second example allow us to combine expressions with **and** and **or** statements.
+
+|Code| Meaning|
+|:-:| :-|
+| `! EXPRESSION` | EXPRESSION is false |
+| `[[ EXPRESSION1 && EXPRESSION2 ]]` | Both EXPRESSION1 and EXPRESSION2 are true |
+| `[[ EXPRESSION1 || EXPRESSION2 ]]` | One or both are true |
+
+
+</div>
+
+<div class = "help">
+
+**Look up instead of memorizing**
+
+Don't worry about trying to memorize these statements. Even people who are writing this kind of code daily regularly have to look up these statements.
+
+It is more important to know what types of commands are possible so that you can envision what you want your code to do. You can always return to this page or type something like "Bash file is empty statement" into a search engine to quickly find what you need.
+
+</div>
+
+### Quiz: Conditional Statements
+
+Without running this code, predict what it will do:
+
+```
+for number in 1 2 3 4 5 6 7
+do
+  if [ $number -gt 4 ]
+    then echo $number is bigger than four
+  elif [ $number -eq 2 ]
+    then echo $number is two
+  elif [ $number -eq 6 ]
+    then echo $number is six
+  elif [ $number -lt 5 ]
+    then echo $number is smaller than five
+  fi
+done
+```
+
+Select all true statements about the output of the above code:
+
+[[X]] The first line will be `1 is smaller than five`
+[[ ]] The second line will be `2 is smaller than five`
+[[X]] The third line will be `3 is smaller than five`
+[[X]] The fourth line will be `4 is smaller than 5`
+[[X]] The fifth line will be `5 is bigger than 4`
+[[ ]] The sixth line will be `6 is six`
+[[X]] The seventh line will be `7 is bigger than 4`
+***
+<div class = "answer">
+
+When the code gets to `2`, it sees the test `[ $number -eq 2 ]` and stops there because it is true. The code never gets to checking whether 2 is less than 5.
+
+When the code gets to `6`, it first sees the test ` [ $number -gt 4 ]`. Since that is true, it never checks whether 6 is equal to 6.
+
+</div>
+***
+
+
+## Additional Resources
+
+Software Carpentry has a series of lessons on the [Unix Shell](https://swcarpentry.github.io/shell-novice/), including one specifically on [loops](https://swcarpentry.github.io/shell-novice/05-loop/index.html).
```

Comment:
We should try to come up with some other resources, since this is tricky. Maybe some cheatsheets... it might be cool to also have a link to something about creating flowcharts for loops (mostly the ones I've found are for Python, but I can keep looking)

Reply:
I found some more to add in here, two cheat sheets, a cute video, and programming language agnostic flow charts!